.github/workflows/gating.yaml

```diff
@@ -101,3 +110,50 @@ jobs:
         ignore: DL3041
         failure-threshold: warning
 
+  build-image:
+    name: Build Cachi2 image and run smoketest on it
+    runs-on: ubuntu-latest
+    steps:
+      - name: add checkout action...
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build Cachi2 image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ${{ env.IMAGE_NAME }}
+          tags: ${{ github.sha }}
+          arch: amd64
+          containerfiles: |
+            ./Containerfile
+
+      - name: Check image created and Cachi2 version
+        run: |
+          buildah images | grep '${{ env.IMAGE_NAME }}'
+          podman run -t ${{ env.IMAGE_NAME }}:${{ github.sha }} --version
+
+      - name: Run smoketest on built images
+        run: |
+          git clone -b ${{ env.TEST_BRANCH }} ${{ env.TEST_REPO }} ${{ env.TEST_FOLDER }}
+          podman run -t -v ${{ env.TEST_FOLDER }}:${{ env.TEST_FOLDER }} \
+            ${{ env.IMAGE_NAME }}:${{ github.sha }} \
+            fetch-deps \
+            --source ${{ env.TEST_FOLDER }} \
+            --output ./cachi2-output \
+            --package '{"path": ".", "type": "gomod"}'
+
+     # Authenticate to container image registry to push the image
+      - name: Podman Login
+        uses: redhat-actions/podman-login@v1
+        if: github.ref == 'refs/heads/main'
+        with:
+          registry: ${{ env.IMAGE_REGISTRY }}
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_PWD }}
+
+      - name: Push image to Quay.io registry
+        if: github.ref == 'refs/heads/main'
+        run: |
+          podman push ${{ env.IMAGE_NAME }}:${{ github.sha }} \
+            ${{ env.IMAGE_REGISTRY }}/${{ env.IMAGE_NAMESPACE }}/${{ env.IMAGE_NAME }}:${{ env.IMAGE_TAG }}
```

Comment:
How are we  going to preserve multiple versions in the registry?

Should we push both
```
quay.io/containerbuildsystem/cachi2:ef0c82bb199a56e8a9ff49b2bbe7dfc0cf00efc1
quay.io/containerbuildsystem/cachi2:latest
```
To give users a chance to pin to a specific commit?

And/or should we do something like operator-manifest, which also [builds on release](https://github.com/containerbuildsystem/operator-manifest/blob/51425a5327fe771e134a97f59183eb652dcb2459/.github/workflows/multiarch-build.yaml#L7) and [pushes the release tag](https://github.com/containerbuildsystem/operator-manifest/blob/51425a5327fe771e134a97f59183eb652dcb2459/.github/workflows/multiarch-build.yaml#L78)?

Reply:
That make sense, let's push both. Might be useful for RCA and debugging.

I haven't thought about releases yet but it does not make sense to wait for anything. +1 I'll add it now. I hope it works the same way. :)

Reply:
I'm thinking whether we want to run the whole pipeline (tests, linters, everything) on release or just the image build. But it probably makes sense to run the whole thing

Reply:
I would say we want to run the whole thing when it comes to release

Reply:
but I can prepare a separate workflow just for the build and testing as operator-manifest has it https://github.com/containerbuildsystem/operator-manifest/blob/51425a5327fe771e134a97f59183eb652dcb2459/.github/workflows/multiarch-build.yaml